overlord/state/state.go

```diff
@@ -373,6 +373,27 @@ func (s *State) Prune(pruneWait, abortWait time.Duration) {
 	}
 }
 
+// ensureBy asks for an ensure pass by the given time (or earlier), immediately if when is in the past. It is a no-op if when.IsZero().
+func (s *State) ensureBy(when time.Time) {
+	if !when.IsZero() {
```

Comment:
This seems error prone, because `EnsureBefore(0)` means immediately rather than ignore. It'd probably be better to get rid of this method entirely, and do this logic and decision inside `task.At` itself.